src/main/java/de/retest/recheck/review/ignore/PixelDiffFilter.java

```diff
@@ -16,9 +16,15 @@
 
 	private static final String PIXEL = "px";
 
+	/**
+	 * Indicates whether {@link #pixelDiff} is specified as float ({@code true}) or integer ({@code false}). Although
+	 * internally it is always treated as a double, this is important for serialization.
+	 */
+	private final boolean specifiedAsFloat;
 	private final double pixelDiff;
 
-	public PixelDiffFilter( final double pixelDiff ) {
+	public PixelDiffFilter( final boolean specifiedAsFloat, final double pixelDiff ) {
```

Comment:
I'm not very fond of these flag arguments (see Clean Code). A solution would either to 
- pass in a specific Formatter (e.g. `DoubleFunction<String> formatter`) - _also not so good_ 
- have a inner class `IntPixelDiffFilter` and `DoublePixelDiffFilter` which uses `PixelDiffFilter` as parent while overwriting `toString` or similar - _introduces verbosity_

What do you think?

Reply:
I see your point, but the "flag argument rule" targets methods/functions, not constructors. I think both are different as constructing/configuring an object can be more complex than calling one of its methods.

In fact, all filters currently violate the SoC principle to a certain degree as they not just know how to filter, but also how they are going to be serialized via the `toString()` method. IMHO having separate (floating and integer) classes or a special formatter isn't worth the effort, but I'm open.

Reply:
> In fact, all filters currently violate the SoC principle to a certain degree

It currently just sticks out (more), since it especially needs special information regarding formatting, although its (logic) implementation does not care about integers.
We can discuss this at a later point, once we do implement SoC.